### ISSUE TITLE
Using pthread with BUILD_AS_WORKER fails

### ISSUE THREAD

**Issue by Matt5sean3** [emscripten-core/emscripten#14881]
Hello,

I ran across a bit of a problem when trying to use the emscripten Worker API along with pthreads. Whenever a thread is called the thread fails with an error. I've been able to use pthread without a problem in other contexts, but when attempting to start a thread from within a worker compiled using BUILD_AS_WORKER and called via the Worker API, starting the thread fails.

I'm running the compiled result in Firefox 90.0.2 on Arch Linux.

## Environment Information
```
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 2.0.27 (7e538a419c3649f3a540a57beab347aa8f6c6271)
clang version 14.0.0 (https://github.com/llvm/llvm-project 78e87970af888bbbd5652c31f3a8454e8e9dd5b8)
Target: wasm32-unknown-emscripten
Thread model: posix
InstalledDir: /home/matthew/Repos/emsdk/upstream/bin
```
## Makefile
```
all: client.js worker.js
CC:=em++

client.js: client.cpp
	$(CC) client.cpp -o client.js

worker.js: worker.cpp
	$(CC) -pthread -s BUILD_AS_WORKER=1 -s PTHREAD_POOL_SIZE=2 worker.cpp -o worker.js

clean:
	rm client.js client.wasm worker.js worker.worker.js worker.wasm
```
## client.cpp
```
#include <emscripten/emscripten.h>
#include <emscripten/html5.h>

worker_handle worker;

void handle_response( char * response, int response_size, void * userdata )
{
  emscripten_console_log( "Received response" );
  response[ response_size - 1 ] = 0;
  emscripten_console_log( response );
  auto char_userdata = reinterpret_cast< char * >( userdata );
  delete[] char_userdata;
}

int main( int argc, char ** argv )
{
  emscripten_console_log( "Creating worker" );
  worker = emscripten_create_worker( "worker.js" );

  char sent_data[ 4 ];
  sent_data[ 0 ] = 0xDE;
  sent_data[ 1 ] = 0xAD;
  sent_data[ 2 ] = 0xBE;
  sent_data[ 3 ] = 0xEF;

  char * char_userdata = new char[ 4 ];
  char_userdata[ 0 ] = 0xDE;
  char_userdata[ 1 ] = 0xAD;
  char_userdata[ 2 ] = 0xBE;
  char_userdata[ 3 ] = 0xEF;

  emscripten_console_log( "Calling worker" );

  emscripten_call_worker( worker, "initialize", sent_data, sizeof( sent_data ), handle_response, char_userdata );

  return 0;
}
```
## worker.cpp
```
#include <emscripten/emscripten.h>
#include <emscripten/html5.h>

#include <thread>

void thread_func( )
{
  emscripten_console_log( "Hello, world!" );
}

extern "C" void EMSCRIPTEN_KEEPALIVE initialize( char * received_data, int received_data_size )
{
  emscripten_console_log( "Received request" );
  std::thread t( thread_func );

  char response_data[ 5 ];
  response_data[ 0 ] = 'A';
  response_data[ 1 ] = 'B';
  response_data[ 2 ] = 'C';
  response_data[ 3 ] = 'D';
  response_data[ 4 ] = 0;

  emscripten_worker_respond( response_data, sizeof( response_data ) );
}
```
## index.html
```
<!DOCTYPE html>
<html>
  <head>
    <title>Test Threads with emscripten's BUILD_AS_WORKER</title>
  </head>
  <body>
    <script type="text/javascript" src="client.js"></script>
    Look at the console!
  </body>
</html>
```
## Firefox Screenshot
![Screenshot from 2021-08-16 12-41-02](https://user-images.githubusercontent.com/7004277/129599012-a4714122-2318-44e6-b224-bbf48eae9639.png)

**Comment by Matt5sean3** [comment]
It would appear that this is a duplicate of the auto-closed issue #8854. The general idea based on those comments seems to be that the Worker API mentioned was going to be deprecated.

I'll look into putting together a PR that either emits an explicit warning if making these APIs both work together is too difficult. Probably at compilation stage emitting a warning if both the `-pthread` and `-s BUILD_AS_WORKER` flags are present ought to do it.

**Comment by Matt5sean3** [comment]
My PR ensured that using both `-pthread` and `-s BUILD_AS_WORKER` will fail compilation when used together and documents this fact, so this issue is closed!
